Add BDD/Unit testing for nginx configuration.

steveworley opened the pull request:
- Adds testing framework as per govcmslagoon.
- Adds no-robots nginx conf.
- Updates CI to run tests.

steveworley committed:
Add BDD/Unit testing for nginx configuration.

- Adds testing framework as per govcmslagoon.
- Adds no-robots nginx conf.
- Updates CI to run tests.

steveworley committed:
Downgrade dependencies for 7.1

steveworley committed:
Remove robots tests as this is added elsewhere.

steveworley committed:
Update config.yml

srowlands commented on .docker/images/nginx/tests/bootstrap.php:
```diff
@@ -0,0 +1,54 @@
+  $response = null;
+  $path = '/' . ltrim($path, '/');
+
+  exec("docker-compose exec -T --env LAGOON_IMAGE_VERSION_PHP=7.1 test curl {$uri}{$path} {$opts} 2>/dev/null", $response);
```
GovCMS8 is PHP7.2 by default

srowlands replied:
Although.. actually what is this env var used for in this context? 

`LAGOON_IMAGE_VERSION_PHP` expects a value like `-v0.23.2`, (as per `.env.default`) - but I'm unclear why this is required?

steveworley committed:
Update bootstrap.php

steveworley committed:
Update PrivateFilesTest.php

steveworley committed:
Merge branch 'develop' into feature/nginx-testing

steveworley committed:
Merge branch 'develop' into feature/nginx-testing

steveworley committed:
Add robots deny and update tests.

steveworley committed:
Merge branch 'feature/nginx-testing' of github.com:govCMS/govcms8lagoon into feature/nginx-testing

simesy commented on docker-compose.yml:
```diff
@@ -22,7 +22,7 @@ services:
         GOVCMS_PROJECT_VERSION: ${GOVCMS_PROJECT_VERSION:-^1.0}
         DRUPAL_CORE_VERSION: ${DRUPAL_CORE_VERSION:-^8.7}
-        LAGOON_IMAGE_VERSION: ${LAGOON_IMAGE_VERSION_PHP}
+        LAGOON_IMAGE_VERSION: ${LAGOON_IMAGE_VERSION_PHP:-}
```
Just checking what this change is since LAGOON_IMAGE_VERSION_PHP was removed from docker-compose.yml in master.

steveworley replied:
Yeah, I'll rebase with master before the merge. It was just to remove some warnings that were annoying me :joy:

steveworley committed:
Add Robots tests to validate nginx behaviour.

- nginx adds X-Robots-Tag depending on the domain to restrict crawlers.
- nginx delivers a robots disallow based on host.

stooit reviewed (changes requested):
Rebased against latest `develop` to strip change to docker-compose

steveworley committed:
Merge branch 'develop' into feature/nginx-testing
